src/ForceTorqueMonitor.c

```diff
@@ -0,0 +1,164 @@
+    rosidl_runtime_c__String__Sequence__init(&g_messages_ForceTorqueMonitor.jointExternalTorque->name, g_Ros_Controller.totalAxesCount); // Number of joints in message
+
+    for (int jointIndex = 0; jointIndex < g_Ros_Controller.totalAxesCount; jointIndex += 1)
+    {
+        rosidl_runtime_c__String__assign(&g_messages_ForceTorqueMonitor.jointExternalTorque->name.data[jointIndex], g_nodeConfigSettings.joint_names[jointIndex]);
+    }
+    rosidl_runtime_c__float64__Sequence__init(&g_messages_ForceTorqueMonitor.jointExternalTorque->position, 0);
+    rosidl_runtime_c__float64__Sequence__init(&g_messages_ForceTorqueMonitor.jointExternalTorque->velocity, 0);
+    rosidl_runtime_c__float64__Sequence__init(&g_messages_ForceTorqueMonitor.jointExternalTorque->effort, g_Ros_Controller.totalAxesCount);
+}
+
+void Ros_ForceTorqueMonitor_Cleanup()
+{
+    rcl_ret_t ret;
+
+    Ros_Debug_BroadcastMsg("Cleanup TCP force torque publisher");
+    ret = rcl_publisher_fini(&g_publishers_ForceTorqueMonitor.forceTorqueTCP, &g_microRosNodeInfo.node);
+    if (ret != RCL_RET_OK)
+        Ros_Debug_BroadcastMsg("Failed cleaning up TCP force torque publisher: %d", ret);
+    geometry_msgs__msg__WrenchStamped__destroy(g_messages_ForceTorqueMonitor.forceTorqueTCP);
+
+    Ros_Debug_BroadcastMsg("Cleanup joint external torque publisher");
+    ret = rcl_publisher_fini(&g_publishers_ForceTorqueMonitor.jointExternalTorque, &g_microRosNodeInfo.node);
+    if (ret != RCL_RET_OK)
+        Ros_Debug_BroadcastMsg("Failed cleaning up joint external torque publisher: %d", ret);
+    sensor_msgs__msg__JointState__destroy(g_messages_ForceTorqueMonitor.jointExternalTorque);
+
+
+}
+
+void Ros_ForceTorqueMonitor_UpdateLocation()
+{
+    LONG status;
+    rcl_ret_t ret;
+
+    //Timestamp
+    INT64 theTime = rmw_uros_epoch_nanos();
+
+    Ros_Nanos_To_Time_Msg(theTime, &g_messages_ForceTorqueMonitor.forceTorqueTCP->header.stamp);
+    Ros_Nanos_To_Time_Msg(theTime, &g_messages_ForceTorqueMonitor.jointExternalTorque->header.stamp);
+
+    MP_IO_INFO registerInfo[6];
+    USHORT registerValues[6];
+    double forceTCPValues[6];
+    double torqueJointValues[6];
+    
+    //assign TCP force data register adresses
+    registerInfo[0].ulAddr = TCP_FORCE_FX_REG_ID;
+    registerInfo[1].ulAddr = TCP_FORCE_FY_REG_ID;
+    registerInfo[2].ulAddr = TCP_FORCE_FZ_REG_ID;
+    registerInfo[3].ulAddr = TCP_FORCE_MX_REG_ID;
+    registerInfo[4].ulAddr = TCP_FORCE_MY_REG_ID;
+    registerInfo[5].ulAddr = TCP_FORCE_MZ_REG_ID;
+
+    //Read TCP force data from registers
+    status = mpReadIO(registerInfo, registerValues, 6);
+
+    if (status != OK)
+    {
+        Ros_Debug_BroadcastMsg("Failed to get TCP force torque data: %u", status);
+    }
+
+    //convert register data to proper format and assign to corresponding wrench fields
+    g_messages_ForceTorqueMonitor.forceTorqueTCP->wrench.force.x = (registerValues[0] - 10000.0) * 0.1;
```

Comment:
Why is 10000 being subtracted? 
Is this for scaling the units? If so, that should be division.

Reply:
The formula taken directly from official manual 
![image](https://github.com/Yaskawa-Global/motoros2/assets/100153601/9c618e5b-3d85-49f4-ab8a-1528c8ea2451)

Reply:
@ted-miller I can also confirm that the uint16 values come from M-registers and those are represented with `10000` for `0`. Therefore, this calculation is also correct for our HC20SDTP robot.

Reply:
that `10000` is in the documentation (also quoted by @iydv), so it's clear it needs to be included in the conversion.